server/src/db.py

```diff
@@ -1,40 +1,103 @@
-import os
+import json
+import logging
+import typing as t
 
-from motor import motor_asyncio
+from aiohttp import web
+from asyncpg.pool import Pool
+from config import BASE_DIR
+from src.encrypt import encrypt_password
+from src.models.game import Game
+from src.models.user import User as ValidUser
+from src.middlewares import User
 
 
-DB_CONNECTION = None
-DATABASE = None
+log = logging.getLogger(__name__)
 
 
-def db_connection():
-    global DB_CONNECTION
-    if DB_CONNECTION is None:
-        DB_CONNECTION = motor_asyncio.AsyncIOMotorClient(os.environ['MONGO_URL'])
-    return DB_CONNECTION
+def get_query(query_name: str) -> t.Optional[str]:
+    try:
+        with open(BASE_DIR / f'queries/{query_name}.sql', 'r') as file:
+            return file.read()
+    except FileNotFoundError:
+        log.error(f'Query {query_name} not found')
 
 
-def db() -> motor_asyncio.AsyncIOMotorDatabase:
-    global DATABASE
-    if DATABASE is None:
-        DATABASE = db_connection()[os.environ['MONGO_DATABASE']]
-    return DATABASE
+async def create_db(app: web.Application):
+    log.info('create_db')
+    pool: Pool = app['pool']
+    async with pool.acquire() as conn:
+        await conn.execute(get_query('init'))
 
 
-async def create_room(data: dict):
-    await db().rooms.insert_one(data)
+async def create_user(pool: Pool, user: ValidUser):
+    user.password = encrypt_password(user.password)
+    async with pool.acquire() as conn:
+        await conn.execute(
+            get_query('create_user'),
+            user.username, user.password
+        )
 
 
-async def get_rooms(page: int, limit: int) -> (list, int):
-    cursor = db().rooms.aggregate([
-        {"$limit": limit},
-        {"$skip": (page - 1) * limit},
-    ])
-    return await cursor.to_list(None), await db().rooms.count_documents({})
+async def create_game(pool: Pool, user: User) -> Game:
+    async with pool.acquire() as conn:
+        game = await conn.fetchrow(
+            get_query('create_game'),
+            user.id
+        )
+    game = Game(*game)
+    game.field = json.loads(game.field)
+    return game
 
 
-async def room_is_occupied(data: dict) -> bool:
-    obj = await db().rooms.find_one({"room_name": data["room_name"]})
-    if obj is None:
-        return False
-    return True
+async def get_user(pool: Pool, user: ValidUser) -> t.Optional[ValidUser]:
+    async with pool.acquire() as conn:
+        user = dict(await conn.fetchrow(
+            get_query('get_user'),
+            user.username
+        ))
+    if user is not None:
+        return ValidUser(user)
+
+
+async def get_game(pool: Pool, id: int) -> t.Optional[Game]:
```

Comment:
actually, id - is reserved name in python, so i think, it can be resolved as _id or gID (game id) or something on your taste